new-lamassu-admin/src/pages/Blacklist/Blacklist.js

```diff
@@ -0,0 +1,141 @@
+
+const useStyles = makeStyles(styles)
+
+const DELETE_ROW = gql`
+  mutation DeleteBlacklistRow($cryptoCode: String, $address: String) {
+    deleteBlacklistRow(cryptoCode: $cryptoCode, address: $address)
+  }
+`
+
+const GET_BLACKLIST = gql`
+  query getBlacklistData {
+    blacklist {
+      cryptoCode
+      address
+    }
+    cryptoCurrencies {
+      display
+      code
+    }
+  }
+`
+
+const SAVE_CONFIG = gql`
+  mutation Save($config: JSONObject) {
+    saveConfig(config: $config)
+  }
+`
+
+const GET_INFO = gql`
+  query getData {
+    config
+  }
+`
+
+const ADD_ROW = gql`
+  mutation InsertBlacklistRow($cryptoCode: String!, $address: String!) {
+    insertBlacklistRow(cryptoCode: $cryptoCode, address: $address) {
+      cryptoCode
+      address
+    }
+  }
+`
+
+const Blacklist = () => {
+  const { data: blacklistResponse } = useQuery(GET_BLACKLIST)
+  const { data: configData } = useQuery(GET_INFO)
+  const blacklistData = R.path(['blacklist'])(blacklistResponse) ?? []
+  const availableCurrencies =
+    R.path(['cryptoCurrencies'], blacklistResponse) ?? []
+
+  const classes = useStyles()
+
+  const [showModal, setShowModal] = useState(false)
+  const [clickedItem, setClickedItem] = useState({
+    code: 'BTC',
+    display: 'Bitcoin'
+  })
+
+  const groupByCode = R.groupBy(obj => obj.cryptoCode)
+  const formattedData = groupByCode(blacklistData)
+
+  const onClickSidebarItem = e => {
+    setClickedItem({ code: e.code, display: e.display })
+  }
+
+  const [deleteEntry] = useMutation(DELETE_ROW, {
+    onError: () => console.error('Error while deleting row'),
+    refetchQueries: () => ['getBlacklistData']
+  })
+
+  const handleDeleteEntry = (cryptoCode, address) => {
+    deleteEntry({ variables: { cryptoCode, address } })
+  }
+
+  const [addEntry] = useMutation(ADD_ROW, {
+    onError: () => console.error('Error while adding row'),
+    refetchQueries: () => ['getBlacklistData']
+  })
+
+  const [saveConfig] = useMutation(SAVE_CONFIG, {
+    refetchQueries: () => ['getData']
+  })
+
+  const isSelected = it => {
+    return clickedItem.code === it.code
+  }
+
+  const toggleModal = () => {
+    setShowModal(!showModal)
+  }
+
+  const addToBlacklist = (cryptoCode, address) => {
+    addEntry({ variables: { cryptoCode, address } })
+  }
+
+  return (
+    <>
+      <TitleSection title="Blacklisted addresses">
+        <Link onClick={toggleModal}>Blacklist new addresses</Link>
+      </TitleSection>
+      <Grid container className={classes.grid}>
+        <Sidebar
+          data={availableCurrencies}
+          isSelected={isSelected}
+          displayName={it => it.display}
+          onClick={onClickSidebarItem}
+        />
+        <div className={classes.content}>
+          <BlacklistTable
+            data={formattedData}
+            selectedCoin={clickedItem}
+            handleDeleteEntry={handleDeleteEntry}
+            saveConfig={saveConfig}
+            configData={configData}
```

Comment:
We can probably simplify this into moving the logic to the outside of the BlacklistTable component. The idea is to make the component as dumb as possible with relation to the context. 

Instead of passing down the configData as a whole you can just pass down rejectAddressReuse and have Blacklist.js figure out the namespacing logic, the same is true for saving.